File: testdata/mock.go
// Package testdata contains test fixtures for hollowcheck.
package testdata

// MockUser represents a user with fake data.
type MockUser struct {
	ID    string
	Email string
	Name  string
}

// GetTestUser returns a user with mock data.
func GetTestUser() *MockUser {
	return &MockUser{
		ID:    "12345",
		Email: "test@example.com",
		Name:  "foo",
	}
}

// GetAnotherUser returns another mock user.
func GetAnotherUser() *MockUser {
	return &MockUser{
		ID:    "00000",
		Email: "user@example.com",
		Name:  "bar",
	}
}

// MockConfig has placeholder values.
type MockConfig struct {
	APIKey   string
	Endpoint string
	Password string
}

// GetMockConfig returns a config with fake data.
func GetMockConfig() *MockConfig {
	return &MockConfig{
		APIKey:   "asdf1234",
		Endpoint: "https://api.example.com/v1",
		Password: "changeme",
	}
}

// GetDescription returns lorem ipsum placeholder text.
func GetDescription() string {
	return "lorem ipsum dolor sit amet"
}

// GetPhoneNumber returns a placeholder phone.
func GetPhoneNumber() string {
	return "xxx-xxx-xxxx"
}

// SequentialIDs returns fake sequential IDs.
func SequentialIDs() []string {
	return []string{"11111", "22222", "33333"}
}

File: testdata/mock.js
// Test fixtures for hollowcheck - mock data patterns.

/**
 * Represents a user with fake data.
 */
class MockUser {
  constructor(id, email, name) {
    this.id = id;
    this.email = email;
    this.name = name;
  }
}

function getTestUser() {
  return new MockUser("12345", "test@example.com", "foo");
}

function getAnotherUser() {
  return new MockUser("00000", "user@example.com", "bar");
}

/**
 * Has placeholder values.
 */
class MockConfig {
  constructor(apiKey, endpoint, password) {
    this.apiKey = apiKey;
    this.endpoint = endpoint;
    this.password = password;
  }
}

function getMockConfig() {
  return new MockConfig(
    "asdf1234",
    "https://api.example.com/v1",
    "changeme"
  );
}

function getDescription() {
  return "lorem ipsum dolor sit amet";
}

function getPhoneNumber() {
  return "xxx-xxx-xxxx";
}

function sequentialIds() {
  return ["11111", "22222", "33333"];
}

module.exports = { MockUser, MockConfig, getTestUser, getAnotherUser, getMockConfig, getDescription, getPhoneNumber, sequentialIds };
